File: backend/PaymentService/src/main/java/com/project/PaymentService/service/PaymentService.java
package com.project.PaymentService.service;

import java.util.List ;
import java.util.Map;


import com.project.PaymentService.entity.OrderDto;
import com.project.PaymentService.entity.OrderResponseDTO;
import com.project.PaymentService.entity.RevenueDTO;
import com.project.PaymentService.entity.WeeklyPaymentDTO;
import com.razorpay.RazorpayException;
import com.project.PaymentService.entity.PaymentDTO;

public interface PaymentService {
	 public Map<String, Object> createOrder(OrderDto dto) throws RazorpayException;
	 OrderResponseDTO createOrder(Long bookingId) throws RazorpayException;
	public boolean verifyPaymentSignature(String razorpay_order_id, String razorpay_payment_id,
			String razorpay_signature) throws RazorpayException;
	
	 // Get all payments
    List<PaymentDTO> getAllPayments();
    
    // Get payments by turf ID
    List<PaymentDTO> getPaymentsByTurfId(Long turfId);
    
    // Get total revenue
    RevenueDTO getTotalRevenue();
    
    // Get revenue by turf ID
    RevenueDTO getRevenueByTurfId(Long turfId);
    
    List<WeeklyPaymentDTO> getWeeklyPaymentStats();
}

File: frontend/TurfNation/src/services/PaymentService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8888';

// Create axios instance
const axiosInstance = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});

// Add request interceptor to include token
axiosInstance.interceptors.request.use(
  (config) => {
    const token = sessionStorage.getItem('token');
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

// Add response interceptor for error handling
axiosInstance.interceptors.response.use(
  (response) => response,
  (error) => {
    if (error.response?.status === 401) {
      sessionStorage.clear();
      window.location.href = '/signin';
    }
    return Promise.reject(error);
  }
);

// Payment Service Functions
export const paymentService = {
  // Get order details for payment
  getOrderDetails: async (bookingId) => {
    try {
      const response = await axiosInstance.get(`/payment/getorder`, {
        params: { bookingId }
      });
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Verify payment after Razorpay callback
  verifyPayment: async (paymentData) => {
    try {
      const response = await axiosInstance.post('/payment/verify', paymentData);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get all payments (Admin)
  getAllPayments: async () => {
    try {
      const response = await axiosInstance.get('/payment/getAllPayments');
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get payments by Turf ID (Admin)
  getPaymentsByTurfId: async (turfId) => {
    try {
      const response = await axiosInstance.get(`/payment/getPaymentsByTurfId/${turfId}`);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get total revenue (Admin)
  getTotalRevenue: async () => {
    try {
      const response = await axiosInstance.get('/payment/getTotalRevenue');
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  },

  // Get revenue by Turf ID (Admin)
  getRevenueByTurfId: async (turfId) => {
    try {
      const response = await axiosInstance.get(`/payment/getRevenueByTurfId/${turfId}`);
      return response.data;
    } catch (error) {
      throw error.response?.data || error.message;
    }
  }
};

export default axiosInstance;

// Export individual functions for direct import
export const getAllPayments = paymentService.getAllPayments;
export const getPaymentsByTurfId = paymentService.getPaymentsByTurfId;
export const getTotalRevenue = paymentService.getTotalRevenue;
export const getRevenueByTurfId = paymentService.getRevenueByTurfId;
